www/js/ui.js

```
@@ -229,15 +235,46 @@ function buildUI(filter_function) {
     }
     updateGfxModImage();
     updateSettingsString();
-    updateForm();
+    checkStoredRom();
+    if (!storedRomArray) updateForm();
 
     ID("rom").onchange = updateForm
 
     ID("submitbutton").onclick = startRomGeneration;
 }
 
+function checkStoredRom()
+{
+    try
+    {
+        var storedRom = localStorage.getItem("ladx_rom");
+        if (storedRom)
+        {
+            var bin = atob(storedRom);
+            var array = new Uint8Array(bin.length);
+            for (var k = 0; k < bin.length; k++)
+            {
+                array[k] = bin.charCodeAt(k);
+            }
+            if (getRomChecksum(array) == 89122269)
+            {
+                romArray = array;
+                storedRomArray = array;
+                setValidRom(true, "ROM has been loaded");
+            }
+            else
+            {
+                localStorage.removeItem("ladx_rom");
+            }
+        }
+    }
+    catch(e){}
```

Empty catch should at least log the error in the js console